lib/fast_jsonapi/object_serializer.rb

```diff
@@ -34,11 +34,13 @@ def initialize(resource, options = {})
     end
 
     def serializable_hash
-      if self.class.is_collection?(@resource, @is_collection)
-        return hash_for_collection
+      result = if self.class.is_collection?(@resource, @is_collection)
+        hash_for_collection
+      else
+        hash_for_one_record
       end
-
-      hash_for_one_record
+      return deep_symbolize_keys(result) if self.class.cache_store_instance
```

Comment:
manually break the batchloader promise to gain some performance benefits. 

cannot simple call as_json because the chowbus response-serializer won't accept hash with string key, and `#with_indifferent_access` shallow symbolized the hash keys.